cocotb/decorators.py

```diff
@@ -460,3 +457,25 @@ def __init__(self, f, timeout=None, expect_fail=False, expect_error=False,
 
     def __call__(self, *args, **kwargs):
         return RunningTest(self._func(*args, **kwargs), self)
+
+
+@public
+def timeout(timeout_time, timeout_unit=None):
+
+    def decorator(f):
+
+        def wrapper(*args, **kwargs):
+            timeout_timer = cocotb.triggers.Timer(timeout_time, timeout_unit)
+            res = yield [timeout_timer, coroutine(f)(*args, **kwargs)]
+            if res is timeout_timer:
+                raise TimeoutError
```

Comment:
I don't think `TimeoutError` exists in python 2, sadly

Reply:
Ahh Python 2, definitely the farthest thing from my mind at 1 AM. So, should a new/separate exception be defined? Or should a definition be conditionally sourced?

```python
if sys.version_info < (3, 3):
    @public
    class TimeoutError(OSError):
        pass
```

Reply:
> Or should a definition be conditionally sourced?

I'd probably compromise on declaring a custom exception type, but derive it from the builtin `TimeoutError` if it exists